### DISCUSSION TITLE
how to create a lsp client to connect to vscode-cpptools?

### DISCUSSION THREAD

**Discussion by zhangzheyu2simple** [microsoft/vscode-cpptools#7948] [Q&A]
I am trying to add some additional code formatting to the cpp file, so I need the AST of the current cpp file.
So, is it possible to create an lsp client to connect to vscode-cpptools and get the AST? If so, is there such a tutorial?

**Comment by sean-mcmanus** [comment]
We don't provide an AST. Is the additional code formatting based on vcFormat or clang-format or is it some new formatting? You could try using the clang AST.
